Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -334,6 +334,18 @@ public void ReportInputContent(FileArtifact artifact, in FileMaterializationInfo
             }
         }
 
+        /// <summary>
+        /// Report the change affected outputs the pip. Only report those produced by the pip.
+        /// </summary>
+        public void ReportSourceChangeAffectedOutputs(AbsolutePath output, PipOutputOrigin origin, bool isFile)
+        {
+            if (origin == PipOutputOrigin.Produced)

```

Comment:
>origin == PipOutputOrigin.Produced [](start = 16, length = 34)

Is it only for pip that gets a cache miss.

Reply:
I think it should only includes the cache miss pip. If a pip is cache hit, I don't see any possible that its output is change affected.